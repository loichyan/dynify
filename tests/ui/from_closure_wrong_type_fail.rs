fn main() {
    let _ = dynify::from_closure(|slot| slot.write(123i32) as &mut dynify::Opaque<u32>);
}
