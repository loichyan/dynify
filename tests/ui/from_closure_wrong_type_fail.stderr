error[E0605]: non-primitive cast: `&mut Opaque<i32>` as `&mut Opaque<u32>`
 --> tests/ui/from_closure_wrong_type_fail.rs:2:41
  |
2 |     let _ = dynify::from_closure(|slot| slot.write(123i32) as &mut dynify::Opaque<u32>);
  |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object
